packages/protobuf/src/lib.ts

```diff
@@ -0,0 +1,155 @@
+        "wrong-type": "Protobuf methods may only accept a named Model as an input",
+        unconvertible: "input parameters cannot be converted to a Protobuf message",
+      },
+    },
+    "unsupported-field-type": {
+      severity: "error",
+      messages: {
+        unconvertible: paramMessage`cannot convert a ${"type"} to a protobuf type (only intrinsic types and models are supported)`,
+        "unknown-intrinsic": paramMessage`no known protobuf scalar for intrinsic type ${"name"}`,
+        "unknown-scalar": paramMessage`no known protobuf scalar for TypeSpec scalar type ${"name"}`,
+        "recursive-map": "a protobuf map's 'value' type may not refer to another map",
+      },
+    },
+    "namespace-collision": {
+      severity: "error",
+      messages: {
+        default: paramMessage`the package name ${"name"} has already been used`,
+      },
+    },
+    "unconvertible-enum": {
+      severity: "error",
+      messages: {
+        default:
+          "enums must explicitly assign exactly one integer to each member to be used in a Protobuf message",
+        "no-zero-first":
+          "the first variant of an enum must be set to zero to be used in a Protobuf message",
+      },
+    },
+    "nested-array": {
+      severity: "error",
+      messages: {
+        default: "nested arrays are not supported by the Protobuf emitter",
+      },
+    },
+    "invalid-package-name": {
+      severity: "error",
+      messages: {
+        default: paramMessage`${"name"} is not a valid package name (must consist of letters and numbers separated by ".")`,
+      },
+    },
+    "illegal-reservation": {
+      severity: "error",
+      messages: {
+        default:
+          "reservation value must be a string literal, uint32 literal, or a tuple of two uint32 literals denoting a range",
+      },
+    },
+    "model-not-in-package": {
+      severity: "error",
+      messages: {
+        default: paramMessage`model ${"name"} is not in a namespace that uses the '@Protobuf.package' decorator`,
+      },
+    },
+    "anonymous-model": {
+      severity: "error",
+      messages: {
+        default: "anonymous models cannot be used in Protobuf messages",
+      },
+    },
+    package: {
+      severity: "error",
+      messages: {
+        "disallowed-option-type": paramMessage`option '${"name"}' with type '${"type"}' is not allowed in a package declaration (only string, boolean, and number values are allowed)`,
+      },
+    },
+  },
+  emitter: { options: EmitterOptionsSchema },
+});
+
+export const { reportDiagnostic } = TypeSpecProtobufLibrary;
+
+export { $onEmit } from "./proto.js";
+
+export type TypeSpecProtobufLibrary = typeof TypeSpecProtobufLibrary;
+
+const keys = [
+  "fieldIndex",
+  "package",
+  "service",
+  "externRef",
+  "stream",
+  "reserve",
+  "message",
+] as const;
+
+export const state = Object.fromEntries(
+  keys.map((k) => [k, Symbol(`@typespec/protobuf.${k}`)])
```

Comment:
Adding that here in case it got lost but updating `Symbol` to `createStateSymbol`